nilearn/image/image.py

```diff
@@ -502,6 +502,52 @@ def swap_img_hemispheres(img):
 
     # create swapped nifti object
     out_img = nibabel.Nifti1Image(img.get_data()[::-1], img.get_affine(),
-        header=img.get_header())
+                                  header=img.get_header())
 
     return out_img
+
+
+def _index_img(imgs, index):
+    """Helper function for index_img and iter_img
+    """
+    return nibabel.Nifti1Image(imgs.get_data()[..., index],
+                               imgs.get_affine(),
+                               header=imgs.get_header())
+
+
+def index_img(imgs, index):
+    """Indexes into a 4D Niimg-like object in the fourth dimension
+
+    Parameters
+    ----------
+    imgs: 4D Niimg-like object
+          See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+
+    index: Any type compatible with numpy array indexing
+           Used for indexing the 4D data array in the fourth dimension
+
+    Returns
+    -------
+    output: nibabel.Nifti1IMage
+            4D image whose data is a subset of the original image data
+
+    """
+    imgs = check_niimgs(imgs)
```

Comment:
accept_3d is False by default.

I haven't looked at the out of bounds error, I have to say. Also I should probably add some tests for it.